src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1043 @@
+"""Module for SOP Classes of Presentation State (PR) IODs."""
+import logging
+import pkgutil
+from typing import Optional, Sequence, Tuple, Union
+
+import numpy as np
+from pydicom import Dataset
+from pydicom.sr.coding import Code
+from pydicom.uid import ExplicitVRLittleEndian
+from pydicom._storage_sopclass_uids import (
+    AdvancedBlendingPresentationStateStorage,
+    ColorSoftcopyPresentationStateStorage,
+    GrayscaleSoftcopyPresentationStateStorage,
+    PseudoColorSoftcopyPresentationStateStorage,
+)
+from pydicom.valuerep import PersonName
+
+from highdicom.base import SOPClass
+from highdicom.content import (
+    ContentCreatorIdentificationCodeSequence,
+    ModalityLUTTransformation,
+    PaletteColorLUTTransformation,
+)
+from highdicom.pr.content import (
+    _add_displayed_area_attributes,
+    _add_equipment_attributes,
+    _add_graphic_group_annotation_layer_attributes,
+    _add_icc_profile_attributes,
+    _add_modality_lut_attributes,
+    _add_palette_color_lookup_table_attributes,
+    _add_presentation_state_identification_attributes,
+    _add_presentation_state_relationship_attributes,
+    _add_softcopy_presentation_lut_attributes,
+    _add_softcopy_voi_lut_attributes,
+    _get_modality_lut_transformation,
+    _get_softcopy_voi_lut_transformations,
+    _get_icc_profile,
+    AdvancedBlending,
+    BlendingDisplay,
+    GraphicLayer,
+    GraphicGroup,
+    GraphicAnnotation,
+    PresentationLUTTransformation,
+    SoftcopyVOILUTTransformation,
+)
+from highdicom.sr.coding import CodedConcept
+from highdicom.uid import UID
+
+
+logger = logging.getLogger(__name__)
+
+
+class GrayscaleSoftcopyPresentationState(SOPClass):
+
+    """SOP class for a Grayscale Softcopy Presentation State (GSPS) object.
+
+    A GSPS object includes instructions for the presentation of a grayscale
+    image by software.
+
+    """
+
+    def __init__(
+        self,
+        referenced_images: Sequence[Dataset],
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        content_label: str,
+        content_description: Optional[str] = None,
+        graphic_annotations: Optional[Sequence[GraphicAnnotation]] = None,
+        graphic_layers: Optional[Sequence[GraphicLayer]] = None,
+        graphic_groups: Optional[Sequence[GraphicGroup]] = None,
+        concept_name: Union[Code, CodedConcept, None] = None,
```

Comment:
It appears this attribute (and a few others) has type 3. Generally we have not been including those as constructor parameters. Do you want to have them here anyways?

Reply:
Agreed, this is somewhat opinionated. I wanted this here as a method by which a user could encode their intent for the particular presentation state in a somewhat standardized way, rather than relying on things like Series Description. There may be multiple presentation states accumulated in a study and it would be good to have a way to identify them.

But we can remove if it's complicating things

Reply:
Thanks for clarifying. I think it would be in principle nice to have additional attributes, but if we keep them we may provide a bit more documentation on how are intended to be used (unfortunately, the standard doesn't really help in this regard).

Reply:
I thought about this for a while and in this case I don't know what else to say. There is not more information given in the standard about how this should be used. I think it's somewhat self explanatory, no?

Reply:
I think it's fine leaving this as is.